src/bindings/python/src/pyopenvino/graph/passes/matcher_pass.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "pyopenvino/graph/passes/matcher_pass.hpp"
+
+#include <pybind11/functional.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include <string>
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/pattern.hpp"
+
+namespace py = pybind11;
+
+void regclass_passes_Matcher(py::module m) {
+    py::class_<ov::pass::pattern::Matcher, std::shared_ptr<ov::pass::pattern::Matcher>> matcher(m, "Matcher");
+    matcher.doc() = "openvino.runtime.passes.Matcher wraps ov::pass::pattern::Matcher";
+    matcher.def(py::init([](const std::shared_ptr<ov::Node>& node, const std::string& name) {
+                    return std::make_shared<ov::pass::pattern::Matcher>(node, name);
+                }),
+                py::arg("node"),
+                py::arg("name"),
+                R"(
+                Creates Matcher object with given pattern root node and matcher name.
+                Matcher object is used for pattern matching on Model.
+
+                :param node: pattern root node.
+                :type node: Node
+
+                :param name: pattern name. Usually matches the MatcherPass class name.
+                :type name: str
+    // )");
+
+    matcher.def(py::init([](ov::Output<ov::Node>& output, const std::string& name) {
+                    return std::make_shared<ov::pass::pattern::Matcher>(output, name);
+                }),
+                py::arg("output"),
+                py::arg("name"),
+                R"(
+                Creates Matcher object with given pattern root node output and matcher name.
+                Matcher object is used for pattern matching on Model.
+
+                :param node: pattern root node output.
+                :type node: Output
+
+                :param name: pattern name. Usually matches the MatcherPass class name.
+                :type name: str
+    // )");
+
+    matcher.def("get_name",
+                &ov::pass::pattern::Matcher::get_name,
+                R"(
+                Get Matcher name.
+
+                :return: Matcher name.
+                :rtype: str
+    //)");
+
+    matcher.def("get_match_root",
+                &ov::pass::pattern::Matcher::get_match_root,
+                R"(
+                Get matched root node inside Model. Should be used after match() method is called.
+
+                :return: matched node.
+                :rtype: Node
+    //)");
+
+    matcher.def("get_match_value",
+                &ov::pass::pattern::Matcher::get_match_value,
+                R"(
+                Get matched node output inside Model. Should be used after match() method is called.
+
+                :return: matched node output.
+                :rtype: Output
```

Comment:
The same comment as for Matcher

Reply:
I used https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/pyopenvino/graph/node.cpp#L288 as a reference.